application/client/src/app/ui/views/sidebar/search/providers/providers.ts

```diff
@@ -500,6 +506,67 @@ export class Providers {
         this.subjects.get().context.emit(event);
     }
 
+
+    public contextMenuOptions(items: IMenuItem[]): IMenuItem[] {
+        const historySession = history.get(this.session);
+        if(historySession === undefined) {
+            this.logger.error('History session is not defined');
+            return items
+        }
+        items.push({ /* Delimiter */ });
+        const store = this.session.search.store();
+        const showExport: boolean = (store.filters().get().length + store.charts().get().length + store.disabled().get().length) !== 0;
+        showExport && items.push(
+        {
+            caption: 'Export All to File',
+            handler: () => this._exportFilters(historySession),
+        });
+        items.push(
+        {
+            caption: 'Import from File',
+            handler: () => this._importFilterFile(historySession),
+        });
+        return items;
+    }
+
+    private _exportFilters = (historySession: HistorySession) => {
+        bridge.files().select.save()
+        .then((filename: string | undefined) => {
+            if (filename === undefined)
+                return;
+            history.export([historySession.collections.uuid], filename);
+        })
+        .catch(error => this.logger.error(error.message));
+    }
+
+    private _importFilterFile = (historySession: HistorySession) => {
```

Comment:
please use regular declaraction of methods:

```
private _importFilterFile(historySession: HistorySession): void {
    ....
}
```
or 
```
protected importFilterFile(historySession: HistorySession): void {
    ....
}
```

this `private _importFilterFile = (historySession: HistorySession) => {}` declaration of property (with function as value), but not declaration of method

But see also comment above, I would group methods `_importFilterFile` and `_exportFilters`